docs/how-to/contribute-code.rst

```diff
@@ -354,26 +354,31 @@ Now you are ready to start hacking on Toga!
 What should I do?
 =================
 
-Start by running the core test suite:
+Start by running the core test suite. Toga uses
+`tox <https://tox.readthedocs.io/en/latest/>`__ to manage the testing process.
+To set up a testing environment and run the core test suite:
 
 .. tabs::
 
   .. group-tab:: macOS
 
     .. code-block:: bash
 
+      (venv) $ pip install tox
```

Comment:
This will work; but it's probably better to modify the install command further up the page, so that you run `pip install -e core[dev]` rather than just `pip install -e core`. That way all the development requirements will be installed, with any version pins that are currently in effect.

Reply:
Sorry, to be honest, I +/- copied this from the `how to contribute code` from `toga-chart` after I realized that the tox installation is missing here, and I don't have any experience with `setuptools`.
Just to be on the safe side:
When I use `pip install -e ./core[dev]` (is this the correct syntax?), then `pip`will read the `dev` section of `setup.cfg` in `[options.extras_require]`which will tell it to install also `pre-commit` and `tox`. So I can also leave out the separate pre-commit installation here. But I still need to describe `pre-commit install` (which will set up pre-commit in git). Is this correct?

Reply:
Sorry for the mislead from the toga-chart docs. However, you've got the idea - adding `[dev]` installs the "extra", which means you don't need the explicit explicit pip install of pre-commit, but you *do* need the `pre-commit install` step to initialise pre-commit. 

This is definitely a case of fixing a bug that you're drawn attention to, rather than being responsible for causing it. This should have been caught when we added the `[dev]` extras definition, so thanks for going the extra distance to fix it for us.